lib/eslint.js

```diff
@@ -617,33 +657,45 @@ module.exports = (function() {
             Object.keys(config.rules).filter(function(key) {
                 return getRuleSeverity(config.rules[key]) > 0;
             }).forEach(function(key) {
-
-                var ruleCreator = rules.get(key),
-                    severity = getRuleSeverity(config.rules[key]),
-                    options = getRuleOptions(config.rules[key]),
-                    rule;
-
-                if (ruleCreator) {
-                    try {
-                        rule = ruleCreator(new RuleContext(
-                            key, api, severity, options,
-                            config.settings, config.ecmaFeatures
-                        ));
-
-                        // add all the node types as listeners
-                        Object.keys(rule).forEach(function(nodeType) {
-                            api.on(nodeType, timing.enabled
-                                ? timing.time(key, rule[nodeType])
-                                : rule[nodeType]
-                            );
-                        });
-                    } catch(ex) {
-                        ex.message = "Error while loading rule '" + key + "': " + ex.message;
-                        throw ex;
+                var ruleCreator,
+                    severity,
+                    options,
+                    rule,
+                    deprecationMsg;
+
+                if (!rules.get(key)) {
```

Comment:
@ilyavolodin As-is, this PR addresses missing rules.  If a rule is missing, it first checks if there is a deprecation notice in `deprecations.json`.  Then it creates a stub rule which uses that message or a generic message that the definition could not be found.  I believe that was the original intent as you describe it and as described in the issue.

I do see @gyandeeps's point about the term "deprecated," which usually is taken to mean it's still there, but you shouldn't use it anymore.  Perhaps there is a better word we could use for the current cases.  Maybe instead of `deprecations.json` we call the file something like `removed-rules.json`?  Would that satisfy everyone for the time being?